lib/archive.ts

```diff
@@ -88,13 +90,53 @@ async function copySourceToDest(
       srcDirPath.push(rootDir);
     }
 
+    const sourceDirs = [];
+
     if (sourceDir) {
-      srcDirPath.push(sourceDir);
+      sourceDirs.push(...(Array.isArray(sourceDir) ? sourceDir : [sourceDir]));
```

Comment:
This might be an unnecessary check, but is there any value in making sure the sourceDirs are unique?

Reply:
Good catch, I'll add a dedupe